client/features/output-visualization-page/components/UndoAndRedo.tsx

```diff
@@ -16,38 +17,26 @@ const useTemporalStore = <T,>(
 
 const UndoAndRedo = () => {
   const { undo, redo } = useTemporalStore((state) => state);
+  const canUndo = useTemporalStore((state) => state.pastStates.length > 0);
+  const canRedo = useTemporalStore((state) => state.futureStates.length > 0);
 
   return (
     <>
       <ButtonGroup orientation='vertical'>
         <IconButton
+          disabled={!canUndo}
           aria-label='undo'
```

Comment:
The aria-label values (`undo`, `redo`) can be more descriptive for better accessibility. For example: `Undo last action` or `Redo last action`.